### PR TITLE
[WEB-1338] Add patient CGM stats to clinic patients list

### PR DESCRIPTION
**Pull request by clintonium-119**
Single PR for the Population Health Summary UI for the following tickets:
[WEB-1338]
[WEB-1565]
[WEB-1566]
[WEB-1567]
[WEB-1568]
[WEB-1570]
[WEB-1616]
[WEB-1617]

**NOTES**

1. The unit tests are not yet complete.  I will post a comment when they are ready for review.  The application code, however, should be very close to it's final form, and I anticipate only minor changes prior to be required as I complete the unit tests
2. There have been some general updates to the table header and pagination UIs for various other pages within the Clinic UI to keep them consistent with the new styles for the update Figma spec in [WEB-1338] for the `ClinicPatients` table.  One big impact was that we are no longer using the pagination built in to the Table component for these view, but rather rendering and controlling it in view state.
3. The UI is still using fake generated data.  I'll remove that portion of code once the real summary data is available on the remotes
4. In the main PR file (`ClinicPatients.js`), there is some commented out code that I've left in for rendering the glycemic events column.  These were initially intended as part of the MVP, but were put on the backburner until they have proper requirement definitions in place.
5. There are also a number of prop-type warning fixes, as they were starting to pile up....

Related PRs
https://github.com/tidepool-org/viz/pull/318
https://github.com/tidepool-org/platform-client/pull/136

[WEB-1338]: https://tidepool.atlassian.net/browse/WEB-1338?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[WEB-1565]: https://tidepool.atlassian.net/browse/WEB-1565?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[WEB-1566]: https://tidepool.atlassian.net/browse/WEB-1566?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[WEB-1567]: https://tidepool.atlassian.net/browse/WEB-1567?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[WEB-1568]: https://tidepool.atlassian.net/browse/WEB-1568?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[WEB-1570]: https://tidepool.atlassian.net/browse/WEB-1570?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[WEB-1616]: https://tidepool.atlassian.net/browse/WEB-1616?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[WEB-1617]: https://tidepool.atlassian.net/browse/WEB-1617?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/pages/clinicworkspace/ClinicPatients.js [hunk 1]
```diff
@@ -133,6 +227,28 @@ export const ClinicPatients = (props) => {
     }
   }, [fetchingPatientsForClinic]);
 
+  React.useEffect(() => {
```
Comment (krystophv) [inline app/pages/clinicworkspace/ClinicPatients.js:230]:
```suggestion
  useEffect(() => {
```

File: app/pages/clinicworkspace/ClinicPatients.js [hunk 2]
```diff
@@ -133,6 +227,28 @@ export const ClinicPatients = (props) => {
...
+
+    // update patientFetchMinutesAgo every minute thereafter
+    const fetchTimeInterval = setInterval(() => {
+      setPatientFetchMinutesAgo(moment.utc().diff(patientFetchMoment, 'minutes'));
+    }, 1000 * 60);
+
+    return () => clearInterval(fetchTimeInterval);
+  }, [clinic?.lastPatientFetchTime]);
+
+  React.useEffect(() => {
```
Comment (krystophv) [inline app/pages/clinicworkspace/ClinicPatients.js:244]:
```suggestion
  useEffect(() => {
```

File: app/pages/clinicworkspace/ClinicPatients.js [hunk 3]
```diff
@@ -144,80 +260,370 @@ export const ClinicPatients = (props) => {
...
-      </Flex>
+      </>
     );
   };
 
+  function handleRefreshPatients() {
+    trackMetric(prefixPopHealthMetric('Refresh data'), { clinicId: selectedClinicId });
+    let fetchOptions = { ...patientFetchOptions };
+    if (isEmpty(fetchOptions.search)) delete fetchOptions.search;
+    dispatch(actions.async.fetchPatientsForClinic.bind(null, api, clinic.id, fetchOptions)());
```
Comment (krystophv) [inline app/pages/clinicworkspace/ClinicPatients.js:612]: I don't _think_ this `.bind` is necessary

File: app/pages/clinicworkspace/ClinicPatients.js [hunk 4]
```diff
@@ -335,6 +741,194 @@ export const ClinicPatients = (props) => {
...
+    const formattedLastUploadReminderTime = selectedPatient?.lastUploadReminderTime && sundial.formatInTimezone(
+      selectedPatient?.lastUploadReminderTime,
+      timePrefs?.timezoneName ||
+        new Intl.DateTimeFormat().resolvedOptions().timeZone,
+      'MM/DD/YYYY [at] h:mm a'
+    );
+
+    return (
+      <Dialog
+        id="resendInvite"
```
Comment (krystophv) [inline app/pages/clinicworkspace/ClinicPatients.js:754]: nit: could use a more accurate `id`

File: app/pages/clinicworkspace/ClinicPatients.js (hunk 3 above)
Comment (clintonium-119) [reply app/pages/clinicworkspace/ClinicPatients.js]: Correct.  Fixed this (and one like it)

File: app/pages/clinicworkspace/ClinicPatients.js (hunk 4 above)
Comment (clintonium-119) [reply app/pages/clinicworkspace/ClinicPatients.js]: updated

**Comment by clintonium-119** [comment]
@krystophv Unit tests are now complete.
I did receive a bit of design feedback that I still need to incorporate.  Basically we want the placement of the main action buttons and/or search bar for some of the other views to match the new placement on the clinic patients view.

I'll ping you when those are up too.  Should be minor

**Comment by clintonium-119** [comment]
@krystophv Considating table header element positioning is complete.  Ready for final PR review.

**Review by krystophv — APPROVED** [review]
Excellent work. All changes since my last set of comments LGTM. :shipit:

**Pull request merged**